cli/util/retry.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use std::future::Future;
+use std::time::Duration;
+
+pub fn retry<F: FnMut() -> Fut, Fut: Future<Output = Result<T, E>>, T, E>(
+  mut f: F,
+) -> impl Future<Output = Result<T, E>> {
+  const ATTEMPTS: usize = 5;
+  const MAX_WAIT: Duration = Duration::from_secs(1);
+  const MIN_WAIT: Duration = Duration::from_micros(1);
+  async move {
+    let mut wait = Duration::from_millis(1);
+    let mut attempt = 1;
+    loop {
+      let result = f().await;
+      if result.is_ok() {
+        return result;
+      }
+      if attempt >= ATTEMPTS {
+        return result;
+      }
+      tokio::time::sleep(wait).await;
```

Comment:
The initial wait seems quite short and waiting a total of 2s (?) seems a bit long when there's a legitimate error. That said, maybe we should just retry once like what's done in file_fetcher.rs, but that's probably not good enough.

Reply:
How about 10ms->50ms->250ms? So three retries, with a worst case of 310ms.

Reply:
I suggest 100->250->500ms. If the service doesn't recover then it's probably down.